### DISCUSSION TITLE
Store assets offline validations

### DISCUSSION THREAD

**Discussion by TrianguloY** [Triple-T/gradle-play-publisher#1070] [Q&A] [answered]
Is there a command/flag to validate the store files without actually uploading them? Things like: check title max length, check locales folder correctness, check screenshots size/format...

My user case is to have a check in Github/Gitlab that runs on commits and PRs, to make sure that the assets are always correct and shouldn't fail when uploaded.

I've tried to search for a command or a flag but couldn't find any. I've also tried to search for an existing action with no avail. Apologize if it does exists and I've missed it.

**Comment by SUPERCILEX** [comment, marked as answer]
We actually used to have some validation within the plug-in, but we removed it because whenever Google changed some requirement, there would be a delay in when the plugin would match that requirement. You also can't guarantee success offline: for example, you might already have images uploaded and try to upload new ones which will fail because you exceed the combined image count limit.

Because it's not possible to guarantee success, I don't think it makes sense to put this in the plugin. However, you could always add something like a python script to do your own validation.

**Reply by TrianguloY** [reply]
Understood, thanks for the explanation.

I have a [simple script in one of my apps](https://github.com/TrianguloY/UrlChecker/blob/master/.github/workflows/validate-store-files.yml) with the title/descriptions length check, I was wondering if there is an official and more advanced checker.

I'll probably extract, polish and publish it as an (unofficial) Github action, maybe it's useful for others, will report it here if that happens.

Thanks!

**Reply by SUPERCILEX** [reply]
Sounds great!

**Comment by TrianguloY** [comment]
As I mentioned, I extracted, polished (a bit) and published it on the GitHub marketplace:
https://github.com/marketplace/actions/triple-t-validation-action

Note: I tried to make sure and explain that it's unofficial (not affiliated with you), but if you think it's not enough or can be confusing just tell me and I'll adapt as necessary.

**Reply by SUPERCILEX** [reply]
This is great, thanks!
